sources/api/controllers/GroupController.ts

```diff
@@ -178,31 +179,64 @@ export async function getMembers(req: Request, res: Response) {
 // Add a User to a Group
 //
 export async function addUsersToGroup(req: Request, res: Response) {
-    // TODO
+
+    const requestingUser = getRequestingUser(req);
+
+    // Recover arguments
+    const groupId: number = req.params.group_id;
+    const newUserPseudos: string[] = req.body.users;
+
+    // Check if user lists is properly defined
+    if (!newUserPseudos || newUserPseudos.length == 0) {
+        throw new ApiException(422, "Missing users argument");
+    }
+
+    // Recover Database connection
+    const conn = await DbConnection.getConnection();
+
+    // Recover User instances for all pseudos
+    const newUsers: User[] = await Promise.all<User>(newUserPseudos.map((pseudo: string) => {
+        return User.findByPseudo(conn, pseudo);
+    }));
+
+    if (newUsers.indexOf(undefined) !== -1) {
+        throw new ApiException(404, `User ${newUserPseudos[newUsers.indexOf(undefined)]} not found`);
+    }
+
+    // Get the requested Group
+    const group: Group = await Group.findById(conn, groupId);
+    if (!group) {
+        throw new ApiException(403, "Group not accessible");
+    }
+
+    // Check if the requesting User can access this Group
+    let calendarToOwner: CalendarsToOwners = await CalendarsToOwners.findCalendarRelation(conn, group.calendar.id, requestingUser.id);
+    if (!calendarToOwner) {
+        throw new ApiException(403, "Group not accessible");
+    }
+
+    for (const user of newUsers) {
+
+        // Check if User is not already in group
+        calendarToOwner = await CalendarsToOwners.findCalendarRelation(conn, group.calendar.id, user.id);
```

Comment:
I wanted to be able to tell which one is already in the group (I forgot to change the message)

Reply:
You can still do this

It will return all the relations that already exist, so just get one (or all) of the corresponding users if you want a clearer error message

My point was more on the insertion part: make sure all can be added, and then add them all


Eventually, for the first part do this to get all conflicting users:
```sql
SELECT * FROM users AS user
    INNER JOIN calendar_to_owner AS cto ON cto.owner_id = user.id
    INNER JOIN groups AS group on group.calendar_id = cto.calendar_id
    WHERE group.id = $group_id
    AND user.pseudo IN ("michel", "romain", "nicole", ...)
;
```